examples/ex_palette.c

```diff
@@ -193,31 +163,42 @@ int main(int argc, char **argv)
          redraw = false;
          al_clear_to_color(al_map_rgb_f(0, 0, 0));
 
-         interpolate_palette(pal, pals[p1 * 2], pals[p2 * 2], pos);
+         al_set_shader_float("pal_set_1", p1 * 2);
+         al_set_shader_float("pal_set_2", p2 * 2);
+         al_set_shader_float("pal_interp", pos);
 
-         al_set_shader_float_vector("pal", 3, pal, 256);
          if (background)
             al_draw_bitmap(background, 0, 0, 0);
 
          for (i = 0; i < 8; i++) {
             Sprite *s = sprite + 7 - i;
             float pos = (1 + sin((t / 60 + s->t) * 2 * ALLEGRO_PI)) / 2;
-            interpolate_palette(pal, pals[s->i], pals[s->j], pos);
-            al_set_shader_float_vector("pal", 3, pal, 256);
+            al_set_shader_float("pal_set_1", s->i);
+            al_set_shader_float("pal_set_2", s->j);
+            al_set_shader_float("pal_interp", pos);
             al_draw_rotated_bitmap(bitmap,
                64, 64, s->x, s->y, s->angle, s->flags);
          }
          
          {
             float sc = 0.5;
-            al_set_shader_float_vector("pal", 3,
-               pals[(int)t % 20 > 15 ? 6 : 0], 256);
+            al_set_shader_float("pal_set_1", (int)t % 20 > 15 ? 6 : 0);
+            al_set_shader_float("pal_interp", 0);
```

Comment:
I'd expect the corner sprites to swap palettes, but they remain the same colors... I'll have to look into this more tomorrow.

Reply:
It's because the 0th and. the 6th palettes were the same colors, both being hue shifted at equivalent values (0 and 360). I change the hue shifter to be not a factor of the period, and now it alters between two colors as expected.